core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -187,11 +187,15 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     assertEquals(1, listener.callsToAssigned)
     assertEquals(0, listener.callsToRevoked)
 
-    Thread.sleep(3500)
+    // as long as we are part of max.poll no rebalance should be triggered
+    Thread.sleep(500)
+    assertEquals(1, listener.callsToAssigned)
 
-    // we should fall out of the group and need to rebalance
-    awaitRebalance(consumer, listener)
-    assertEquals(2, listener.callsToAssigned)
+    // after we extend longer than max.poll a rebalance should be triggered
+    // NOTE we need to have a relatively much larger value than max.poll to let heartbeat expired for sure
+    Thread.sleep(3000)
```

Comment:
Can we get rid of this `sleep` since we're blocking below anyway?

Reply:
Unfortunately we cannot since we need to wait for this amount of time "without polling", and then we should poll and see the rebalance being triggered.